jsonnet/kube-prometheus/components/prometheus.libsonnet

```
@@ -32,16 +32,17 @@ local defaults = {
   mixin:: {
     ruleLabels: {},
     _config: {
+      local _config = self,
       prometheusSelector: 'job="prometheus-' + defaults.name + '",namespace="' + defaults.namespace + '"',
       prometheusName: '{{$labels.namespace}}/{{$labels.pod}}',
       // TODO: remove `thanosSelector` after 0.10.0 release.
-      thanosSelector: '',
+      thanosSelector: 'job="thanos-sidecar"',
       thanos: {
         targetGroups: {
           namespace: defaults.namespace,
         },
         sidecar: {
-          selector: 'job="thanos-sidecar"',
+          selector: _config.thanosSelector,
```

```suggestion
          selector: defaults.mixin._config.thanosSelector,
```

This way `local _config = self` is not needed in L35 and we have full visibility into where the configuration came from